haskell/cabal.bzl

```diff
@@ -643,8 +643,7 @@ build times, and does not require drafting a `.cabal` file.
 """,
 )
 
-def _haskell_cabal_binary_impl(ctx):
-    hs = haskell_context(ctx)
+def _haskell_cabal_binary_or_test_impl(ctx, hs, is_test, name):
```

Comment:
```suggestion
def _haskell_cabal_binary_or_test_impl(ctx, hs, is_test, component_name):
```
To avoid confusion with `hs.label.name`, which the implementation uses in various places.